be/src/olap/tablet_reader.cpp

```diff
@@ -214,7 +214,7 @@ Status TabletReader::_capture_rs_readers(const ReaderParams& read_params) {
             // unique keys with merge on write, no need to merge sort keys in rowset
             need_ordered_result = false;
         }
-        if (_aggregation) {
+        if (_aggregation || _direct_mode) {
```

Comment:
current condition is confusing
```suggestion
if (_direct_mode) {
    need_ordered_result = false;
}
```